### PR TITLE
Dashboard: Hide in progress template actions 

### PR DESCRIPTION
**Pull request by BrittanyIRL**
## Summary
Hide template actions that are still in progress with a flag. This includes the template filters, sort, and search.

Screen Shot 2020-06-11 at 2 41 51 PM

## Relevant Technical Choices
- adds a new flag called `enableInProgressTemplateActions` that hides actions on explore templates that are not connected to an API yet. 
- Used the `onClick` prop that gets passed to the Header and Body Options (respectively) to check if that UI should be rendered. So when `enableInProgressTemplateActions` is false, that prop is `false` instead of passing in a function. I chose this because if that function isn't there the UI to trigger that function is useless anyways. It seemed streamlined to me. 

## User-facing changes
- You should no longer see the following UI on explore templates: 
  - Search Input
  - Filter Dropdowns  
  - Sort Dropdown 

## Testing Instructions
- See that the above UI is not showing up in the explore templates view (and storybook Dashboard/Views/ExploreTemplates/Header/Default
- Flip the `enableInProgressTemplateActions` to true and see that the UI is back to "normal" with nothing hidden. 

---

Addresses #2381

### CODE REVIEW

File: assets/src/dashboard/app/views/exploreTemplates/header/index.js [hunk 1]
```diff
@@ -105,14 +104,40 @@ function Header({ filter, totalTemplates, search, templates, sort, view }) {
...
+        {TemplateFilters}
       </PageHeading>
       <BodyViewOptions
         resultsLabel={resultsLabel}
         layoutStyle={view.style}
         handleLayoutSelect={view.toggleStyle}
         currentSort={sort.value}
         pageSortOptions={TEMPLATES_GALLERY_SORT_MENU_ITEMS}
-        handleSortChange={sort.set}
+        handleSortChange={enableInProgressTemplateActions && sort.set}
```
Comment (carloskelly13) [inline assets/src/dashboard/app/views/exploreTemplates/header/index.js]: When `enableInProgressTemplateActions` is `false` the `&&` operator will short circuit and return `false` which is incompatible with the propType.

```
Warning: Failed prop type: Invalid prop `handleSortChange` of type `boolean` supplied to `BodyViewOptions`, expected `function`.
```

Removing the `isRequired` and setting this to `enableInProgressTemplateActions ? sort.set : undefined` would work.

File: assets/src/dashboard/app/views/exploreTemplates/header/index.js [hunk 2]
```diff
@@ -105,14 +104,40 @@ function Header({ filter, totalTemplates, search, templates, sort, view }) {
...
+
+  return (
+    <Layout.Squishable>
+      <PageHeading
+        centerContent
+        defaultTitle={__('Templates', 'web-stories')}
+        searchPlaceholder={__('Search Templates', 'web-stories')}
+        stories={templates}
+        handleTypeaheadChange={
+          enableInProgressTemplateActions && search.setKeyword
```
Comment (carloskelly13) [inline assets/src/dashboard/app/views/exploreTemplates/header/index.js]: When enableInProgressTemplateActions is false the && operator will short circuit and return false which is incompatible with the propType.

(Look at the comment at line 140 to see the similar problem and solution)

File: assets/src/dashboard/app/views/exploreTemplates/header/test/header.js [hunk 3]
```diff
@@ -158,11 +161,35 @@ describe('Explore Templates <Header />', function () {
...
+          }}
+          view={{
+            style: VIEW_STYLE.GRID,
+            pageSize: { width: 200, height: 300 },
+          }}
+        />
+      </LayoutProvider>,
+      { enableInProgressTemplateActions: false }
+    );
+    expect(queryAllByRole('input')).toHaveLength(0);
```
Comment (carloskelly13) [inline assets/src/dashboard/app/views/exploreTemplates/header/test/header.js]: This test creates a false positive as written. The role for inputs are `textbox`.

When this is `expect(queryAllByRole('textbox')).toHaveLength(0);`

Then the test correctly works depending on if the flag is false, then length 0, true, then length is 1.

**Review by carloskelly13 — DISMISSED** [review]
This is really good with just a couple edge-case nits!

**Comment by BrittanyIRL** [comment]
@googlebot I consent

File: assets/src/dashboard/app/views/exploreTemplates/header/test/header.js (hunk 3 above)
Comment (BrittanyIRL) [reply assets/src/dashboard/app/views/exploreTemplates/header/test/header.js]: thanks for fixing this

**Comment by carloskelly13** [comment]
@googlebot I consent.

File: assets/src/dashboard/app/views/exploreTemplates/header/index.js [hunk 4]
```diff
@@ -77,16 +83,9 @@ function Header({ filter, totalTemplates, search, templates, sort, view }) {
...
-    <Layout.Squishable>
-      <PageHeading
-        centerContent
-        defaultTitle={__('Templates', 'web-stories')}
-        searchPlaceholder={__('Search Templates', 'web-stories')}
-        stories={templates}
-        handleTypeaheadChange={search.setKeyword}
-        typeaheadValue={search.keyword}
-      >
+  const TemplateFilters = useMemo(() => {
```
Comment (dmmulroy) [inline assets/src/dashboard/app/views/exploreTemplates/header/index.js]: Nit: I think `useMemo` is overkill here and it might make more sense to do something like 

```jsx
const TemplateFilters = enableInProgressTemplateActions ? <Components /> : null;
```

or adding it to the return with `{enableInProgressTemplateActions && <Components />}`

File: assets/src/dashboard/app/views/exploreTemplates/header/index.js [hunk 5]
```diff
@@ -105,14 +104,42 @@ function Header({ filter, totalTemplates, search, templates, sort, view }) {
...
       </PageHeading>
       <BodyViewOptions
         resultsLabel={resultsLabel}
         layoutStyle={view.style}
         handleLayoutSelect={view.toggleStyle}
         currentSort={sort.value}
         pageSortOptions={TEMPLATES_GALLERY_SORT_MENU_ITEMS}
-        handleSortChange={sort.set}
+        handleSortChange={
+          enableInProgressTemplateActions ? sort.set : undefined
```
Comment (dmmulroy) [inline assets/src/dashboard/app/views/exploreTemplates/header/index.js]: Could we change `undefined` here and on line `128` to be a noop function just for extra safety?

**Review by dmmulroy — APPROVED** [review]
Just some minor nits that don't impact functionality at all.  Works as expected 🚀

**Review by dmmulroy — APPROVED** [review]
